notebooks/timeseries.ipynb

```diff
@@ -0,0 +1,568 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# OpenSCM Timeseries\n",
+    "\n",
+    "In OpenSCM, the class for handling timeseries is `openscm.timeseries.Timeseries`. \n",
+    "\n",
+    "A timeseries in OpenSCM is always initialised from arrays i.e. discrete data which are associated with specific points in time. In this notebook we call the initialisation values the 'source data' and the times with which they are associated the 'source times'.\n",
+    "\n",
+    "To estimate what the value of the variable under consideration is at points in time other than the source points (call these points the 'target times'), i.e. to interpolate or extrapolate, we always take the following steps:\n",
+    "\n",
+    "1. start with the data which was originally used to set the timeseries\n",
+    "1. convert the discrete data to continuous data using an interpolation/extrapolation technique\n",
+    "    - e.g. linear interpolation, cubic spline, moving average\n",
+    "1. use the continuous data to determine the values at the target times aka the 'target values'\n",
+    "\n",
+    "There is one additional complication in this process. It is the different conventions that are used with timeseries. For example, sometimes the data value is the value only at the exact point in time for which it is supplied while other times the data value represents the average of the underlying continuous data set over a given time period. These conventions affect how the conversion between continuous and discrete data is done and hence are vital to document and understand.\n",
+    "\n",
+    "*Note:* In order to recover the original data from the target data, you need to know which assumptions have been made in the interpolation process and what convention is being used.\n",
```

Comment:
Do we need this note?

Reply:
I don't know. I found it helpful as I always it should be easy to go back and forth but have only just realised it's not. Maybe it's obvious to everyone except me, in which case I'm happy to remove it.

Reply:
I would remove it, but have no strong feelings about that ;)

Reply:
@rgieseke thoughts? Do we keep this note about why going back and forth isn't that trivial or delete?